### PR TITLE
scheduler_v2: util for implementing agent

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- We need to add some basic utilities for implementing and testing the scheduler agent.

### What is changed and how it works?
- Added some mocked structs for testing.
- Added `checkpointSender`.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/scheduler/agent_mock.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// ProcessorMessenger implements how messages should be sent to the owner,
+// and should be able to know whether there are any messages not yet acknowledged
+// by the owner.
+type ProcessorMessenger interface {
+	// FinishTableOperation notifies the owner that a table operation has finished.
+	FinishTableOperation(ctx context.Context, tableID model.TableID) (bool, error)
+	// SyncTaskStatuses informs the owner of the processor's current internal state.
+	SyncTaskStatuses(ctx context.Context, running, adding, removing []model.TableID) (bool, error)
+	// SendCheckpoint sends the owner the processor's local watermarks, i.e., checkpoint-ts and resolved-ts.
+	SendCheckpoint(ctx context.Context, checkpointTs model.Ts, resolvedTs model.Ts) (bool, error)
```
Comment (3AceShowHand) [inline cdc/scheduler/agent_mock.go:30]: why need to return a `bool`  here to indicate something, can you explain the potential reason behind it?

**Review by 3AceShowHand — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**